src/utils.cpp

```diff
@@ -112,39 +112,28 @@ String& trim(String& str) {
 }
 
 static bool is_lowercase(const String& str) {
-  if (str.empty()) return true;
-
-  char c = str[0];
-  if (!(c >= 'a' && c <= 'z')) return false;
+  return str.end() == std::find_if(str.begin(), str.end(), ::isupper);
```

Comment:
This changes the semantics. `is_lowercase()` is a misnomer here it's more like "is this a valid lowercase CQL identifier?" that doesn't need to be quoted.  `::isupper` would let invalid chars through e.g. `-`, `.`. Also, `_` can't be the first char.

Reply:
Well, arguably this function does not have to worry about if it is a valid cql identifier, cassandra will do that validation, i.e. all we do here is check if any upper case chars, then will add quotes.

And if it does check for the above, what it suppose to do about it?

Does the driver validate the whole CQL statement? probably not.

Reply:
It can be reverted I think, but I do not see it making any real difference, apart from being overcomplicated with no reason.

Reply:
@mpenick any thoughts, do you want this reverted?